src/Mod/TechDraw/Gui/DrawGuiUtil.cpp

```diff
@@ -73,7 +73,7 @@ void DrawGuiUtil::loadArrowBox(QComboBox* qcb)
     qcb->clear();
     int i = 0;
     for (; i < ArrowPropEnum::ArrowCount; i++) {
-        qcb->addItem(tr(ArrowPropEnum::ArrowTypeEnums[i]));
+        qcb->addItem(QCoreApplication::translate("ArrowPropEnum", ArrowPropEnum::ArrowTypeEnums[i]));
```

Comment:
If the strings are translated in the definition of the enum, they don't need to be retranslated here(?).

I didn't think it was possible to translate the data inside a variable.  This could be:
qcb->addItem(ArrowPropEnum::ArrowTypeEnums[i]);

Reply:
The strings aren't translated in the enum definition: that array contains the original English language `const char *` data, the `QT_TRANSLATE_NOOP` function is a literal NOOP as far as the C++ compiler is concerned. The macro expands to the original string argument and nothing else. But `lupdat`e sees it what it scans the source code and extracts those strings. So now when translate is called, even though `lupdate` can't extract anything from the translate call (because it doesn't contain a string literal), the actual real translate function call still occurs at runtime, and the correct string is looked up in the database. This preserves the original English text being stored in the FCStd files, while still giving the user a translated interface.